File: 백준/Gold/1456. 거의 소수/거의 소수.py
from sys import stdin
input = stdin.readline

def eratosthenes():
    num = [True] * (limit+1)
    num[0],num[1] = False,False
    primes = []
    for i in range(2,limit+1):
        if num[i]:
            primes.append(i)
            for j in range(i**2,limit+1,i):
                num[j] = False
    return primes

limit = 10 ** 7
primes = eratosthenes()
a,b = map(int,input().split())
ans = 0
for p in primes:
    i = p
    while i <= b // p:
        ans += (i*p >= a)
        i *= p
print(ans)

File: 백준/Gold/1456. 거의 소수/거의 소수.java
import java.io.*;
import java.util.*;

public class Main {
	static long A;
	static long B;
	static final int limit = 10000000;

	public static void main(String[] args) throws Exception {
		//System.setIn(new FileInputStream("res/input.txt"));
		BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
		StringTokenizer st = new StringTokenizer(br.readLine());
		A = Long.parseLong(st.nextToken());
		B = Long.parseLong(st.nextToken());
		boolean[] num = new boolean[limit + 1];
		for (int i = 2; i <= Math.sqrt(limit + 1); i++) {
			if (!num[i]) {
				for (int j = i * i; j <= limit; j += i) {
					num[j] = true;
				}
			}
		}
		int ans = 0;
		for (int i = 2; i <= limit; i++) {
			if (!num[i]) {
				long cur = i;
				while ((double) i <= (double) B / (double)cur) {
					if ((double) i >= (double) A / (double)cur) {
						ans++;
					}
					cur *= i;
				}
			}
		}
		System.out.println(ans);
	}
}